Modules/Quotas/src/Quotas.Application/Identities/Queries/ListQuotasForIdentity/Handler.cs

```diff
@@ -0,0 +1,63 @@
+using Backbone.BuildingBlocks.Application.Abstractions.Infrastructure.UserContext;
+using Backbone.DevelopmentKit.Identity.ValueObjects;
+using Backbone.Modules.Quotas.Application.DTOs;
+using Backbone.Modules.Quotas.Application.Infrastructure.Persistence.Repository;
+using Backbone.Modules.Quotas.Domain.Aggregates.Identities;
+using Backbone.Modules.Quotas.Domain.Metrics;
+using MediatR;
+
+namespace Backbone.Modules.Quotas.Application.Identities.Queries.ListQuotasForIdentity;
+
+public class Handler : IRequestHandler<ListQuotasForIdentityQuery, ListQuotasForIdentityResponse>
+{
+    private readonly IIdentitiesRepository _identitiesRepository;
+    private readonly MetricCalculatorFactory _metricCalculatorFactory;
+    private readonly IMetricsRepository _metricsRepository;
+    private readonly IdentityAddress _identityAddress;
+
+    public Handler(IUserContext userContext, IIdentitiesRepository identitiesRepository, IMetricsRepository metricsRepository, MetricCalculatorFactory metricCalculatorFactory)
+    {
+        _identityAddress = userContext.GetAddress();
+        _identitiesRepository = identitiesRepository;
+        _metricsRepository = metricsRepository;
+        _metricCalculatorFactory = metricCalculatorFactory;
+    }
+
+    public async Task<ListQuotasForIdentityResponse> Handle(ListQuotasForIdentityQuery request, CancellationToken cancellationToken)
+    {
+        var identity = await _identitiesRepository.Find(_identityAddress, cancellationToken) ?? throw new NotFoundException(nameof(Identity));
+
+        var metrics = await _metricsRepository.FindAll(cancellationToken);
+
+        var individualQuotasForIdentityTasks = identity.IndividualQuotas
+            .Select(async q =>
+            {
+                var calculator = _metricCalculatorFactory.CreateFor(q.MetricKey);
+                var usage = await calculator.CalculateUsage(q.Period.CalculateBegin(), q.Period.CalculateEnd(), identity.Address, cancellationToken);
+
+                var metric = new MetricDTO(metrics.Single(m => m.Key == q.MetricKey));
+
+                return new QuotaDTO(q, metric, usage);
+            });
+
+        var individualQuotasForIdentity = await Task.WhenAll(individualQuotasForIdentityTasks);
+
+        var tierQuotasForIdentityTasks = identity.TierQuotas
+            .Select(async q =>
+            {
+                var calculator = _metricCalculatorFactory.CreateFor(q.MetricKey);
+                var usage = await calculator.CalculateUsage(q.Period.CalculateBegin(), q.Period.CalculateEnd(), identity.Address, cancellationToken);
+
+                var metric = new MetricDTO(metrics.Single(m => m.Key == q.MetricKey));
+
+                return new QuotaDTO(q, metric, usage);
+            });
+
+        var tierQuotasForIdentity = await Task.WhenAll(tierQuotasForIdentityTasks);
+
+        var quotasForIdentity = individualQuotasForIdentity.Concat(tierQuotasForIdentity).ToList();
```

Comment:
I omitted this because it seems to cause this error:
```
System.InvalidOperationException: 'A key cannot be configured on 'IndividualQuota' because it is a derived type. The key must be configured on the root type 'Quota'. If you did not intend for 'Quota' to be included in the model, ensure that it is not referenced by a DbSet property on your context, referenced in a configuration call to ModelBuilder, or referenced from a navigation on a type that is included in the model.'
```

Reply:
This is easy to fix. Just exclude the property in the EntityTypeConfiguration. Because by default EF Core maps every public property to a column of the database.

Reply:
I tried with `builder.Ignore(x => x.AllQuotas);` but it didn't work - the `MessagesRepositoryTests` failed with the same message as before.

Reply:
I guess you didn't recreate the migrations after adding the `builder.Ignore`?

Reply:
Or rather the compiled models. You can run the `compile_models.ps1` script directly.